org.nuspell.Nuspell.json

```diff
@@ -0,0 +1,82 @@
+{
+  "app-id": "org.nuspell.Nuspell",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "19.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "nuspell",
+  "modules" : [
+    {
+      "name": "boost",
+      "buildsystem": "simple",
+      "build-commands": [
+        "./bootstrap.sh --prefix=/app --with-libraries=locale",
+        "./b2 install"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.bz2",
+          "sha256": "9807a5d16566c57fd74fb522764e0b134a8bbe6b6e8967b83afefd30dcd3be81"
+        }
+      ]
+    },
+    {
+      "name": "launcher",
+      "buildsystem": "simple",
+      "build-commands": [
+        "mkdir -p /app/share/applications/",
+        "cat <<EOF >/app/share/applications/org.nuspell.Nuspell.desktop \
```

Comment:
Not sure why you're not shipping this as a separate file?

Reply:
I'm very new to Flatpak. For the command-line application and library we offer, a desktop file is not needed. We added it to AppImage, because it was required there, but that is another story. I know Flatpak works differently. I read in the documentation that a desktop file should/could be added "in order to identify and display available applications". So that was my motivation to add it.

As you indicated, the desktop file should not be in the Flathub repo. First I moved it to a downloadable source some place else. But to minimize the number of downloads, I refactored it to on the fly generation.

If a desktop file not required by Flatpak, I'm glad to remove it all together. If it is good to keep it to display available applications, that is fine with me, as that helps users to see that the installation worked. If you prefer that I add it as a downloadable source, I can do that to. Please advise on this what is the common way to do this. Thanks.

Reply:
It is preferred that metainfo and desktop files are maintained in the upstream's code repository. They are not Flatpak-specific, and if one day some distribution decides to package your application, they'll just re-use these files instead of writing worse version of it. The advantage also is that you can manage it as part of your release process instead of having to remember to update them here.

If for whatever reason you don't think it's a good idea, you can still keep it as a separate file (not inline in the manifest) like you have done with `org.nuspell.Nuspell.metainfo.xml`.

Reply:
Done